plugins/tileacq.py

```diff
@@ -245,16 +381,23 @@ def _move_to_tile(self, idx, orig_pos, tile_size):
         # |
         # --->-->-->--Z
         overlap = 1 - self.overlap.value / 100
-        tile_pos = (orig_pos["x"] + idx[0] * tile_size[0] * overlap,
-                    orig_pos["y"] - idx[1] * tile_size[1] * overlap)
+        # don't move on the axis that is not supposed to have changed
+        main_data = self.main_app.main_data
+        prev_pos = main_data.stage.position.value
+        if numpy.argmax(abs(numpy.subtract(idx, prev_idx))) == 0:
+            tile_pos = (orig_pos["x"] + idx[0] * tile_size[0] * overlap,
+                        prev_pos["y"])
+        else:
+            tile_pos = (prev_pos["x"],
+                        orig_pos["y"] - idx[1] * tile_size[1] * overlap)
 
         logging.debug("Moving to tile %s at %s m", idx, tile_pos)
         f = self.main_app.main_data.stage.moveAbs({"x": tile_pos[0], "y": tile_pos[1]})
         try:
```

Comment:
Assume some kind of very slow speed (eg, 10µs/s), and compute a timeout from it, instead of always hardcoding to 10s.